py17track/profile.py

```
@@ -108,3 +109,70 @@ async def summary(self, show_archived: bool = False) -> dict:
+            API_URL_BUYER,
+            json={
+                "version": "1.0",
+                "method": "AddTrackNo",
+                "param": {"TrackNos": [tracking_number]},
+            },
+        )
+
+        _LOGGER.debug("Add package response: %s", add_resp)
+
+        if add_resp.get("Code") != 0:
+            return False
+
+        if friendly_name is not None:
+            packages = await self.packages()
+            new_package = next(
+                filter(lambda p: (p.tracking_number == tracking_number), packages), None
+            )
+
+            if new_package is not None:
+                _LOGGER.debug(
+                    "Found internal ID of recently added package: %s", new_package.id
+                )
+
+                success = await self.set_friendly_name(new_package.id, friendly_name)
+                if success is False:
+                    _LOGGER.warning(
+                        "Couldn't set friendly name for % (ID: %s)",
+                        tracking_number,
+                        new_package.id,
+                    )
+            else:
+                _LOGGER.warning(
+                    "Couldn't find internal ID of recently added package (%s)",
+                    tracking_number,
+                )
+
+        return True
+
+    async def set_friendly_name(self, internal_id: str, friendly_name: str) -> bool:
+        """Set a friendly name to an already added tracking number.
+
+        internal_id is not the tracking number, it's the ID of an existing package.
+        """
+        remark_resp: dict = await self._request(
+            "post",
+            API_URL_BUYER,
+            json={
+                "version": "1.0",
+                "method": "SetTrackRemark",
+                "param": {"TrackInfoId": internal_id, "Remark": friendly_name},
+            },
+        )
+
+        _LOGGER.debug("Set friendly name response: %s", remark_resp)
+
+        if remark_resp.get("Code") != 0:
```

Don't swallow errors.